test/SimpleSwap.behavior.js

```diff
@@ -1,931 +1,1471 @@
-                      shouldSubmitChequeissuer(unsignedCheque, sender)  
+                    context('when there is no liquidBalance to cover the cheque', function () {
+                      describe(describeTest + 'shouldSubmitCheque', function () {
+                        const sender = alice
+                        shouldSubmitCheque(unsignedCheque, sender)
+                      })
                     })
                   })
-                  context('when we send more than one cheque', async function() {
-                    shouldSubmitChequeissuer(unsignedCheque, sender)
-                    context('when the serial number is increasing', function() {
-                      let secondSerial = new BN(parseInt(unsignedCheque.serial) + 1)
-                      let increasing_serial_unsignedCheque = Object.assign({}, defaultCheque, {serial: secondSerial, signee: defaultCheque.beneficiary})
-                      shouldSubmitChequeissuer(increasing_serial_unsignedCheque, sender)
+                  context('when we send more than one cheque', async function () {
+                    const sender = alice
+                    shouldSubmitCheque(unsignedCheque, sender)
+                    context('when the serial number is increasing', function () {
+                      describe(describeTest + 'shouldSubmitCheque', function () {
+                        const secondSerial = new BN(parseInt(unsignedCheque.serial) + 1)
+                        const increasing_serial_unsignedCheque = Object.assign({}, defaultCheque, { serial: secondSerial, signee: signees })
+                        shouldSubmitCheque(increasing_serial_unsignedCheque, sender)
+                      })
                     })
-                    context('when the serial number stays the same', function() {
-                      let secondSerial = new BN(parseInt(unsignedCheque.serial))
-                      let same_serial_unsignedCheque = Object.assign({}, defaultCheque, {serial: secondSerial, signee: defaultCheque.beneficiary})
-                      it('reverts', async function() {
-                        this.signedCheque = await signCheque(this.simpleSwap, same_serial_unsignedCheque)
-                        await expectRevert(this.simpleSwap.submitChequeissuer(
-                          this.signedCheque.beneficiary,
-                          this.signedCheque.serial, 
-                          this.signedCheque.amount, 
-                          this.signedCheque.timeout,
-                          this.signedCheque.signature, {from: sender}), "SimpleSwap: invalid serial")
+                    context('when the serial number stays the same', function () {
+                      describe(describeTest + 'shouldNotSubmitCheque', function () {
+                        const secondSerial = new BN(parseInt(unsignedCheque.serial))
+                        const same_serial_unsignedCheque = Object.assign({}, defaultCheque, { serial: secondSerial, signee: signees })
```

Comment:
Any particular reason why you use `Object.assign` explicitly instead of `{ ...defaultCheque, serial: secondSerial, signee: signees }`?

Reply:
I found Object.assign to be clearer, but apart from that no strong argumentation. What would be your argumentation for using the syntax above?

Reply:
Both are fine as long as we consistently use the same one. I used a lot of `...` in the old test suite but if you want we can use `Object.assign` from know on.